compiler/src/model/metamodel.ts

```diff
@@ -378,12 +378,30 @@ export class Deprecation {
   description: string
 }
 
+export class Availability {
+  stack?: StackAvailability
+  serverless?: ServerlessAvailability
+}
+
+export class StackAvailability {
+  since?: string
+  featureFlag?: string
+  stability?: Stability
+  visibility?: Visibility
+}
+
+export class ServerlessAvailability {
```

Comment:
I would make a single `Availability` class that also has `since` and `featureFlag`. In the context of serverless:
* `since` may be used for documentation purposes to indicate when a feature was added
* I'm pretty sure there will be some `featureFlag`s at some point. This is very common in SaaS products to provide early access to upcoming features to selected customers. In terms of code generation, this can materialize as custom builds.

Note BTW that we don't have any `feature_flag` visibility defined currently, so we may also remove it altogether (including in `Visibility`) and consider that APIs guarded by a feature-flag on the server are just `experimental` in the API specification.

Reply:
I'm not sure about the name `featureFlag` for serverless, but I agree with the sentiment that we will eventually have some concept of "invitation only" / "opt-in" / "pre-release" features. But since I don't know which of those concepts we will use (possibly more than 1), my inclination is that we shouldn't try to plan for it now.